<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Hi_OpencvClass</class>
 <widget class="QMainWindow" name="Hi_OpencvClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1065</width>
    <height>581</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hi_Opencv</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="bianyuanjiance" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>1051</width>
      <height>531</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="whatsThis">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;dede&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>图像模糊</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>9</x>
        <y>9</y>
        <width>77</width>
        <height>376</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>60</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_blur">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>blur</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>60</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_gaussian">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>gaussian</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>60</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_3">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>median</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>60</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_4">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Bilateral</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>92</x>
        <y>9</y>
        <width>331</width>
        <height>381</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label">
         <property name="maximumSize">
          <size>
           <width>50</width>
           <height>20</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="text">
          <string>input</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_10">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_in">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>10</y>
        <width>321</width>
        <height>381</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="maximumSize">
          <size>
           <width>50</width>
           <height>20</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="text">
          <string>output</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_12">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_out">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>760</x>
        <y>10</y>
        <width>186</width>
        <height>381</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>13</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="pushButton_open">
           <property name="font">
            <font>
             <family>楷体</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Open</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_start">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="font">
            <font>
             <family>楷体</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>start</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="font">
          <font>
           <family>楷体</family>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="text">
          <string>图像模糊</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_6">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>60</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_w">
           <property name="font">
            <font>
             <family>楷体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>像素宽度w：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_w"/>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_7">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>60</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_h">
           <property name="font">
            <font>
             <family>楷体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>像素高度h：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_h"/>
         </item>
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_8">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>60</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_i">
           <property name="font">
            <font>
             <family>楷体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>内核大小：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_i"/>
         </item>
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>图像平滑</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>边缘检测</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_blur_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Sobel导数</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_blur_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Laplace 算子</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_blur_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Canny 检测</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_blur_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>霍夫线变换</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_blur_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>270</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>寻找轮廓</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_blur_7">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>330</y>
        <width>91</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>轮廓矩</string>
      </property>
     </widget>
     <widget class="QScrollArea" name="scrollArea">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>30</y>
        <width>351</width>
        <height>321</height>
       </rect>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>349</width>
         <height>319</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QScrollArea" name="scrollArea_2">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>30</y>
        <width>351</width>
        <height>321</height>
       </rect>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>349</width>
         <height>319</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="pushButton_open_2">
      <property name="geometry">
       <rect>
        <x>850</x>
        <y>30</y>
        <width>88</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Open</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_open_3">
      <property name="geometry">
       <rect>
        <x>940</x>
        <y>30</y>
        <width>88</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>楷体</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>start</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionblur">
   <property name="text">
    <string>blur</string>
   </property>
  </action>
  <action name="actiongaussian">
   <property name="text">
    <string>gaussian</string>
   </property>
  </action>
  <action name="actionmedian">
   <property name="text">
    <string>median</string>
   </property>
  </action>
  <action name="actionBilateral">
   <property name="text">
    <string>Bilateral</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>bianyuanjiance</class>
   <extends>QTabWidget</extends>
   <header>bianyuanjiance.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Hi_Opencv.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Hi_OpencvClass</receiver>
   <slot>on_median_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>560</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_blur</sender>
   <signal>clicked()</signal>
   <receiver>Hi_OpencvClass</receiver>
   <slot>on_blur_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>17</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_gaussian</sender>
   <signal>clicked()</signal>
   <receiver>Hi_OpencvClass</receiver>
   <slot>on_gaussian_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_open</sender>
   <signal>clicked()</signal>
   <receiver>Hi_OpencvClass</receiver>
   <slot>on_open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>773</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>729</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>Hi_OpencvClass</receiver>
   <slot>on_Bilateral_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>425</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>558</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_start</sender>
   <signal>clicked()</signal>
   <receiver>Hi_OpencvClass</receiver>
   <slot>on_start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>939</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>797</x>
     <y>555</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_open()</slot>
  <slot>on_start()</slot>
  <slot>on_blur_show()</slot>
  <slot>on_gaussian_show()</slot>
  <slot>on_median_show()</slot>
  <slot>on_Bilateral_show()</slot>
 </slots>
</ui>
